File: DaNang_2324/Bai1/TONG.py
input_path = "./THCS/2024/DaNang/Bai1/TONG.INP"
output_path = "./THCS/2024/DaNang/Bai1/TONG.OUT"

def calculate_sum_of_largest(input_file, output_file):
    with open(input_file, 'r') as infile:
        # Đọc N và K
        N, K = map(int, infile.readline().strip().split())
        
        # Đọc dãy số
        numbers = list(map(int, infile.readline().strip().split()))

    # Sắp xếp dãy số theo thứ tự giảm dần và tính tổng K phần tử lớn nhất
    largest_sum = sum(sorted(numbers, reverse=True)[:K])

    # Ghi kết quả vào tệp
    with open(output_file, 'w') as outfile:
        outfile.write(f"{largest_sum}\n")

calculate_sum_of_largest(input_path, output_path)

File: THCS/2023_2024/DaNang_2324/Bai1/TONG.cpp
#include <iostream>
#include <fstream>
#include <vector>
#include <algorithm>

using namespace std;

int main() {
    ifstream infile("TONG.INP");
    ofstream outfile("TONG.OUT");
    
    // Đọc N và K
    int N, K;
    infile >> N >> K;

    vector<int> numbers(N);
    
    // Đọc dãy số
    for (int i = 0; i < N; i++) {
        infile >> numbers[i];
    }

    // Sắp xếp dãy số theo thứ tự giảm dần
    sort(numbers.begin(), numbers.end(), greater<int>());
    
    // Tính tổng K phần tử lớn nhất
    int largest_sum = 0;
    for (int i = 0; i < K; i++) {
        largest_sum += numbers[i];
    }

    outfile << largest_sum << endl;

    infile.close();
    outfile.close();

    return 0;
}
